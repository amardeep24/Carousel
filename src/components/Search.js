import React from 'react'
import PropTypes from 'prop-types';
import TextField from '@material-ui/core/TextField';
import Grid from '@material-ui/core/Grid';
import {debounce} from '../utils/utils';
import SearchIcon from '@material-ui/icons/Search';
import { SearchContainer} from './Styled';

function Search({ search }) {
    return (
        <SearchContainer>
            <Grid container spacing={1} alignItems="center">
                <Grid item>
                    <TextField
                        id="outlined-basic"
                        label="Search"
                        variant="outlined"
                        onChange={(e) => debounce(search, e.target.value)()} />
                </Grid>
                <Grid item>
                    <SearchIcon />
                </Grid>
            </Grid>

        </SearchContainer>
    )
}

export default Search

Search.propTypes = {
    search: PropTypes.string
}