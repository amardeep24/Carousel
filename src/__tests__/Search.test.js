import React from 'react';
import { mount, configure } from 'enzyme';
import Search from '../components/Search';
import TextField from '@material-ui/core/TextField';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';

configure({ adapter: new Adapter() });

jest.useFakeTimers();

describe('Test 1::The search component should match the snapshot', () => {
    let component;

    beforeEach(() => {
        component = mount(<Search
            search={(() => console.log("searched"))}
        />);
    });

    it('should match snapshot', () => {
        expect(component).toMatchSnapshot();
    });
});

describe('Test 2::The search component shall call the debounce function', () => {
    let component;
    let spy;
    beforeEach(() => {
        spy =  jest.fn();
        component = mount(<Search
            search={spy}
        />);
    });

    it('should call the callback after debouncing', () => {
        component
        .find(TextField)
        .props()
        .onChange({ target: { value: "mobile" } });
        setTimeout(() => {
            expect(spy).toHaveBeenCalled();
        }, 3000);
        jest.runAllTimers();
    });
});