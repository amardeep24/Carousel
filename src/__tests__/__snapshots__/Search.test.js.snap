// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test 1::The search component should match the snapshot should match snapshot 1`] = `ReactWrapper {}`;
